woocommerce/woocommerce-ios
# Remove legacy code for store creation

Closes #12442 


## Description

We removed the entry point to store creation a few months ago but haven't gotten around to removing the legacy code. This PR removes store creation-related code to avoid having to maintain it later (e.g. when migrating to Swift 6).

Changes include:
- Removed all code in the Store Creation folder and related unit tests.
- Updated `AuthenticationManager`, `AppCoordinator`, `StorePickerCoordinator`, `StorePickerViewController`, `SubscriptionView` and `SubscriptionViewModel` to remove store creation / free trial related logic.
- Removed theme installing logic in `DashboardViewModel`.
- Removed `onUpgradePlan` closure in `StoreOnboardingCoordinator`.

## Testing steps

- Smoke test the app's login and store picker flows to ensure everything still works as expected.
- Optionally smoke test subscription and dashboard screens as well.


## Screenshots

N/A

---
